source/frontend/src/features/mapSideBar/disposition/tabs/offersAndSale/dispositionSale/hooks/useCalculateNetProceeds.tsx

```diff
@@ -17,12 +22,28 @@ export const useCalculateNetProceeds = (isGstEligible: boolean) => {
 
   useEffect(() => {
     if (!isSubmitting) {
-      const saleCosts = realtorCommissionAmount + totalCostOfSale + netBookAmount;
-      const proceedsBeforeSPP = finalSaleAmount - saleCosts;
-      const proceedsAfterSPP = proceedsBeforeSPP - sppAmount;
+      if (isGstEligible) {
+        let calculatedGst;
+        if (gstTouched) {
+          calculatedGst = gstAmount;
+        } else {
+          calculatedGst = gstDecimal ? finalSaleAmount - finalSaleAmount / (1 + gstDecimal) : 0;
+        }
 
-      setFieldValue('netProceedsBeforeSppAmount', proceedsBeforeSPP);
-      setFieldValue('netProceedsAfterSppAmount', proceedsAfterSPP);
+        const saleCosts = calculatedGst + realtorCommissionAmount + totalCostOfSale + netBookAmount;
+        const proceedsBeforeSPP = finalSaleAmount - saleCosts;
+        const proceedsAfterSPP = proceedsBeforeSPP - sppAmount;
+
+        setFieldValue('netProceedsBeforeSppAmount', proceedsBeforeSPP);
+        setFieldValue('netProceedsAfterSppAmount', proceedsAfterSPP);
+      } else {
+        const saleCosts = realtorCommissionAmount + totalCostOfSale + netBookAmount;
+        const proceedsBeforeSPP = finalSaleAmount - saleCosts;
+        const proceedsAfterSPP = proceedsBeforeSPP - sppAmount;
+
+        setFieldValue('netProceedsBeforeSppAmount', proceedsBeforeSPP);
+        setFieldValue('netProceedsAfterSppAmount', proceedsAfterSPP);
```

Comment:
looks like the netproceeds are the same depending on the salecost adding the calulated gst. Thoughts on moving the duplicatad logic after the if statement?

Reply:
Updated